core/state_processor_rip7560.go

```diff
@@ -101,8 +101,8 @@ func BuyGasRip7560Transaction(st *types.Rip7560AccountAbstractionTx, state vm.St
 
 	chargeFrom := *st.Sender
 
-	if len(st.PaymasterData) >= 20 {
-		chargeFrom = [20]byte(st.PaymasterData[:20])
+	if st.Paymaster.Cmp(common.Address{}) != 0 {
```

Comment:
better do `st.Paymaster != nil` 
(and also when parsing a tx, "nillify" the fields - that is remove zero address)

Reply:
I am not doing the parsing, the parsing is done by the libraries, so `0x0000..000` may slip in sometimes - I did add the `nil` check is well. I will keep the zero address check as well for now.